### PR TITLE
Adding test_provider_crud.py back

### PR DESCRIPTION
**Pull request by jdemon519**
Purpose or Intent
=================

test_provider_crud.py somehow got deleted, adding it back

{{pytest: cfme/tests/containers/test_provider_crud.py -v --use-provider container }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 0cc8386c069243eac16ff48a531d215059589128:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/tests/containers/test_containers_properties.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    pytest.mark.uncollectif(
+        lambda: current_version() < "5.6"),
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.tier(1)]
+pytest_generate_tests = testgen.generate(
+    testgen.container_providers, scope="function")
+
+# CMP-9911 # CMP-9877 # CMP-9867
+
+
```
Comment (psav) [inline cfme/tests/containers/test_containers_properties.py]: As you are navigating to the page showing _all_ pods, I think you either want to
a) use the ```has_no_container_providers``` fixture to remove all providers before you run the test, probably meaning you need to define some other fixtures, to get the dependency right
b) Navigate to Pods (through) the provider, so tat you only get pods that are related to the provider you are using.

File: cfme/tests/containers/test_relationships_tables.py [hunk 2]
```diff
@@ -180,8 +166,7 @@ def test_images_rel(provider, rel):
         obj = Image(name, provider)
 
         val = obj.get_detail('Relationships', rel)
-        if val == '0' or val == 'Unknown image source':
-            continue
+        assert val != 'Unknown image source'
```
Comment (psav) [inline cfme/tests/containers/test_relationships_tables.py]: Before it was allowed to continue if the val was equal to "Unknown image source", now we say it absolutely can't be?

File: cfme/tests/containers/test_one_image_analysis.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        On the Images page ---> put a checkmark next to the image
+        Click on Configuration ---> Perform SmartState Analysis
+        Verify the relevant notification is displayed
+        Settings ---> Tasks ---> All VM and Container Analysis Tasks
+        Verify the image is being scanned
+        Wait for the scan to finish ---> Delete All
+    """
+    sel.force_navigate('containers_images')
+    m = 'Analysis successfully initiated'
+    sel.check(image.checkbox)
```
Comment (psav) [inline cfme/tests/containers/test_one_image_analysis.py]: This shouldn't be here, but rather a method on the ```Image``` class should be created called ```run_smartstate_analysis```. In this way others can benefit from this too. The ```force_navigate``` at the beginning of the test and lines 35-40 should be part of this new method too. This test then becomes two steps:

```
image.run_smartstate_analysis()
is_task_finished()
```

File: cfme/tests/containers/test_one_image_analysis.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        'Configuration',
+        'Perform SmartState Analysis',
+        invokes_alert=True)
+    sel.handle_alert()
+    flash.assert_message_contain(m)
+
+    is_task_finished()
+
+
+def is_task_finished():
```
Comment (psav) [inline cfme/tests/containers/test_one_image_analysis.py]: I'm presuming there is no, identifying information to tell which image analysis is complete?
Regardless, this function belongs in the image.py, and not in the test.

**Comment by psav** [comment]
This PR needs squashing

File: cfme/tests/containers/test_containers_properties.py (hunk 1 above)
Comment (jdemon519) [inline cfme/tests/containers/test_containers_properties.py]: @psav I am trying to go with the second option (navigate to Pods through the provider), but what's the best way to do that?
existing_pods = provider.mgmt.list_container_group()  ?

File: cfme/tests/containers/test_relationships_tables.py (hunk 2 above)
Comment (jdemon519) [inline cfme/tests/containers/test_relationships_tables.py]: @psav I will reference the bug that was opened in the script

**Comment by dajoRH** [comment]
Lint report for commit a36ea7bb861149861cd10d6e889cf56b19697395:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit adacb3d06ee378199101aa6be7362e23c640ac7e:

`cfme/tests/containers/test_one_image_analysis.py`:
- :red_circle: Line 6:1: **F401** *'flash' imported but unused*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9d298b9f1923d92b02787102d57b07d44b67e1ba:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/containers/image.py [hunk 5]
```diff
@@ -22,6 +25,18 @@
 )
 
 
+def run_smart_state_analysis(self):
```
Comment (psav) [inline cfme/containers/image.py]: Can this run against a single image?
It seems to me that this should be a classmethod on Image
Possibly even two methods, one that says ```run_smart_state_analysis```
And the other would be a class method that was called ```run_smart_state_analysis_all_images``` or something like that.
Regardless, I think this should be a method of the Image class, perhaps we need a mixin for SSA at some point

File: cfme/tests/containers/test_containers_properties.py (hunk 1 above)
Comment (psav) [inline cfme/tests/containers/test_containers_properties.py]: well doing that will get you the list via mgmtsystem which will be outside of CFME. Presumably you want the list of pods associated with the provider from inside CFME, ie "what CFME says". In this case you probably want to go to the details page and click on one of the relationship links. The summary attribute of the provider should help you out here.

File: cfme/containers/image.py (hunk 5 above)
Comment (jdemon519) [inline cfme/containers/image.py]: Right, this can be run against a single image. I agree about the mixin for SSA, it's just not available right now.

Comment (psav) [inline cfme/containers/image.py]: Right, so lets make it a method on Image, and we can make the SSA mixin later

Comment (psav) [inline cfme/containers/image.py]: Add a ```# TODO``` comment on it that we need to make the mixin

**Comment by jkrocil** [comment]
Description of this PR has **very** little to do with its content. You are adding 10 files with various content and description only says that you're re-adding a previously removed test. I'm not reviewing this until you describe what you're trying to do here.

I'd suggest (as I already have) to split this PR into 2.
The first adding the removed test (which could be merged immediately and I wrongly believed that to be a priority for you - to have a working smoke test) and the other adding additional tests that need reviewing, unlike the CRUD test.

**Pull request closed without merging**